strategy/to_json/from_object_containing.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
+
+import { JSONObject, JSONValue, Serializable } from "../../serializable.ts";
+
+export function fromObjectContaining(
+  value: Record<string, Serializable | Serializable[]>,
+): JSONObject {
+  const record: Record<string, JSONValue> = {};
+  for (const prop in value) {
+    record[prop] = Array.isArray(value[prop])
+      ? (value[prop] as Serializable[]).map((v) => v.tsSerialize())
+      : (value[prop] as Serializable).tsSerialize();
```

Comment:
I thought this already happened by default. Maybe just for arrays? I figure if we have the objects in memory, we know if they are Serializable, so the default strategy should be to serialize _really hard_. Recursively go through each value in an embedded array or object and `tsSerialize()` it.

But again I thought we already did that by default (at least one level deep). Maybe we only did it for arrays?

Reply:
we do this by default with toJSON but that returns a string and parsing causes escaping, by using `tsSerialize` we are avoiding that issue.